main/HSSF/Model/RecordOrderer.cs

```diff
@@ -310,6 +310,9 @@ private static bool IsDVTPriorRecord(RecordBase rb)
             {
                 return true;
             }
+            if (rb is not Record)
+                return false;
+            
             short sid = ((Record)rb).Sid;
```

Comment:
I'd recommend using pattern matching to cast within the condition rather than checking the type and casting later.

Here's what I suggest:
```csharp
if (rb is Record record)
{
    short sid = record.Sid;
    switch (sid)
    ...
}
return false;
```

**Benefits:**
- Avoids explicit cast, making the code type-safe
- Clearer intent and better scoped variable